redbot/cogs/mutes/mutes.py

```diff
@@ -0,0 +1,1378 @@
+                        _("Automatic unmute"),
+                        until=None,
+                    )
+                else:
+                    for channel, mute_data in channels.items():
+                        self._unmute_tasks[f"{channel}{user}"] = asyncio.create_task(
+                            self._auto_channel_unmute_user(guild.get_channel(channel), mute_data)
+                        )
+
+    async def _auto_channel_unmute_user_silent(
+        self, channel: discord.abc.GuildChannel, data: dict
+    ):
+        """This is meant for mass channel unmute without sending the modlog case"""
+        delay = data["until"] - datetime.now(timezone.utc).timestamp()
+        if delay < 1:
+            delay = 0
+        await asyncio.sleep(delay)
+        try:
+            member = channel.guild.get_member(data["member"])
+            author = channel.guild.get_member(data["author"])
+            if not member:
+                return
+            success = await self.channel_unmute_user(
+                channel.guild, channel, author, member, _("Automatic unmute")
+            )
+            if success["success"]:
+                async with self.config.channel(channel).muted_users() as muted_users:
+                    if str(member.id) in muted_users:
+                        del muted_users[str(member.id)]
+            else:
+                chan_id = await self.config.guild(channel.guild).notification_channel()
+                notification_channel = channel.guild.get_channel(chan_id)
+                if not notification_channel:
+                    return
+                await notification_channel.send(
+                    _(
+                        "I am unable to mute {user} in {channel} for the following reason:\n{reason}"
+                    ).format(user=member, channel=channel.mention, reason=success["reason"])
+                )
+        except discord.errors.Forbidden:
+            return
+
+    async def _auto_channel_unmute_user(self, channel: discord.abc.GuildChannel, data: dict):
+        """This is meant to unmute a user in individual channels"""
+        delay = data["until"] - datetime.now(timezone.utc).timestamp()
+        if delay < 1:
+            delay = 0
+        await asyncio.sleep(delay)
+        try:
+            member = channel.guild.get_member(data["member"])
+            author = channel.guild.get_member(data["author"])
+            if not member:
+                return
+            success = await self.channel_unmute_user(
+                channel.guild, channel, author, member, _("Automatic unmute")
+            )
+            if success["success"]:
+                await modlog.create_case(
+                    self.bot,
+                    channel.guild,
+                    datetime.now(timezone.utc),
+                    "cunmute",
+                    member,
+                    channel.guild.me,
+                    _("Automatic unmute"),
+                    until=None,
```

Comment:
You should pass channel here.
```suggestion
                    until=None,
                    channel=channel,
```